content/en/os/1.14.x/concepts/bootstrap-containers/_index.markdown

```diff
@@ -0,0 +1,88 @@
+- the boot process pauses while the bootstrap containers start, run, and exit,
+- any bootstrap container configured to `{{< ver-ref project="os" page="/api/settings/bootstrap-containers#name_mode">}}mode=once{{< /ver-ref >}}` will change to `{{< ver-ref project="os" page="/api/settings/bootstrap-containers#name_mode">}}mode=off{{< /ver-ref >}}` once complete,
+- any bootstrap container set to `{{< ver-ref project="os" page="/api/settings/bootstrap-containers#name_essential">}}essential=true{{< /ver-ref >}}` that exits with an non-zero exit code halts the boot process,
+- unless stopped in the previous step, the boot process continues by loading the orchestrator agent ({{% ver-ref project="os" page="/concepts/components#container-and-orchestrator-support"%}} `kubelet` on `*-k8s-*` {{% /ver-ref %}} variants or the {{% ver-ref project="os" page="/concepts/components#ecs-variants" %}}ECS agent on `*-ecs-*`{{% /ver-ref %}} variants).
+
+### Examples
+
+{{< twocol
+    containerclass="td-max-width-on-larger-screens docs-figure"
+    rowclass="docs-figure" >}}
+    {{< twocol_inner colsplit="7" >}}
+        {{< bootstrap_containers_diagram example=1 >}}
+    {{</ twocol_inner >}}
+    {{% twocol_inner %}}
+This example shows four bootstrap containers (`B1`,`B2`,`B3`, and `B4`) starting.
+The start order of these containers is not sequential.
+Container `B2` runs the longest and the next stage of the boot does not proceed until it completes.
+    {{%/ twocol_inner %}}
+{{</ twocol >}}
+
+{{< twocol
+    containerclass="td-max-width-on-larger-screens docs-figure"
+    rowclass="docs-figure" >}}
+    {{< twocol_inner  colsplit="7" >}}
+        {{< bootstrap_containers_diagram example=2 >}}
+    {{</ twocol_inner >}}
+    {{% twocol_inner %}}
+This example is the same as the previous one except container `B2` is configured with {{< ver-ref project="os" page="/api/settings/bootstrap-containers#name_essential">}}`essential=true`{{< /ver-ref >}}.
+Container `B2` exits with a non-zero exit code and consquently the boot halts.
+    {{%/ twocol_inner %}}
+{{</ twocol >}}
+
+{{< twocol
+    containerclass="td-max-width-on-larger-screens docs-figure"
+    rowclass="docs-figure" >}}
+    {{< twocol_inner  colsplit="7" >}}
+        {{< bootstrap_containers_diagram example=3 >}}
+    {{</ twocol_inner >}}
+    {{% twocol_inner %}}
+This example is the same as the first except container `B4` is configured with {{< ver-ref project="os" page="/api/settings/bootstrap-containers#name_mode">}}`mode=once`{{< /ver-ref >}}.
+After all the bootstrap containers finish, Bottlerocket updates container `B4`'s `mode` to `off`.
+    {{%/ twocol_inner %}}
+{{</ twocol >}}
+
+### Considerations
+
+As a consequence of this lifecycle, you should keep a few things in mind when using bootstrap containers:
+
+1. Bootstrap containers should be short-running and cleanly exit when complete otherwise the node will hang.
```

Comment:
Not completely true, the host will only hang when a bootstrap container that is `essential=true` exits with a non-zero status.

Reply:
This confuses me. If I have a bootstrap container with `sleep infinity` the node will never finish booting, right? Or is the problem with 'clean exit' meaning something different to you?

Reply:
I think the current phrasing is fine. Even if it isn't `essential=true`, we should encourage users to set up bootstrap containers to be both of those attributes - short running and cleanly exiting.

`sleep infinity` would be bad, but I think the point is a bootstrap container that does not have `essential=true` can exit with a failure code, but the node will still continue to boot. So only failed essential containers will cause the node to hang. We could call out that difference, but I think it's good to recommend cleanly exiting all of the time to be safe.

Reply:
Based on @stmcginnis's feedback, I'll keep the wording the same. 

However, I think **I** might be missing something. See @arnaldo2792's [response above](https://github.com/bottlerocket-os/bottlerocket-project-website/pull/257#discussion_r1320241541):

> Systemd will run many services in parallel, but it won't move to the next target (that is multi-user.target) until all the bootstrap containers succeed

If systemd never moves to the next target, what state does that leave the node in? Will it connect to the orchestrator agent?

Reply:
It won't run the orchestrator, but that's not the only service that is left to be started. There might be some other service (I don't exactly know which services) that may still be running, or about to be run, that don't depend on bootstrap containers.

Reply:
Thanks @arnaldo2792! That's what I suspected.